test/integration/api/services_controller_test.rb

```diff
@@ -199,6 +218,7 @@ class SettingsTest < self
     def update_params(oidc_id: nil)
       @update_params ||= { service:
         { intentions_required: '0',
+          deployment_option: 'service_mesh_istio',
```

Comment:
With this line I wanted to test that we can actually change the Deployment option, also that the Istio option was available and that everything was working as expected.

Reply:
I think it is incorrect to set it globally like that. I see tests that expect it to be apicast. I don't know how they pass. But either case, in your test case, you can do:
```ruby
      put admin_service_path(service), params: update_params(deployment_option: 'service_mesh_istio')
```

Then check value is what you expect. For example you can do this on line 93 instead of setting it directly.